Source/WebCore/loader/CookieJar.cpp

```diff
@@ -63,6 +64,26 @@ IncludeSecureCookies CookieJar::shouldIncludeSecureCookies(const Document& docum
     return (url.protocolIs("https"_s) && !document.foundMixedContent().contains(SecurityContext::MixedContentType::Active)) ? IncludeSecureCookies::Yes : IncludeSecureCookies::No;
 }
 
+SameSiteInfo CookieJar::sameSiteInfo(const Document& document, const URL& url, IsForDOMCookieAccess isAccessForDOM)
+{
+    auto frame = document.frame();
+    if (!frame)
+        return { };
+
+    if (frame->loader().client().isRemoteWorkerFrameLoaderClient()) {
+        auto domain = document.isTopDocument() ? RegistrableDomain(document.securityOrigin().data()) : RegistrableDomain(document.siteForCookies());
```

Comment:
This check doesn't make sense, if this is a Remote worker, then the document is always a top document. Service workers only have one dummy Document.